javalib/src/main/scala/java/util/HashMap.scala

```diff
@@ -0,0 +1,187 @@
+package java.util
+
+import java.{util => ju}
+
+import scala.collection.{mutable => mu}
+
+import scala.collection.JavaConversions._
+
+class HashMap[K, V] protected (inner: mu.Map[Box[K], V],
+       initialCapacity: Int, loadFactor: Float) extends AbstractMap[K, V]
```

Comment:
`initialCapacity` and `loadFactor` are completely useless in this primary constructor. Remove them, and perform the validity checks in the constructors that do take this parameters.